vllm/worker/habana_model_runner.py

```diff
@@ -984,7 +984,7 @@ def execute_model(
             execute_model_kwargs.update({"bypass_hpu_graphs":not use_graphs})
         htorch.core.mark_step()
         # Sample the next token based on previous logits if any.
-        if self.scheduler_config.enable_delayed_sampling and not is_prompt:
+        if self.scheduler_config.enable_delayed_sampling and self.is_driver_worker and not is_prompt:
```

Comment:
@jkaniecki from line 1019 execute_model_kwargs["input_ids"] = output.sampled_token_ids, the rank 0 input_ids is updated, but not for the non-rank 0, would this cause issue? should the input_ids be broadcasted to other ranks?